### ISSUE TITLE
Nested multiline comments not supported

### ISSUE THREAD

**Issue by genius257** [DcodingTheWeb/EasyCodeIt#5]
```autoit3
#cs
	#cs
		nested comment
	#ce
#ce

```

result:
```
---### TOKEN ###---
Type: Comment
Data:
        #cs
                nested comment

---### TOKEN ###---
Type: Directive
Data: ce
```

This is one of those things AutoIt3 does different from many others. Normally the output is expected to act like this, but AutoIt3 needs the same amount of multi-line-comment end blocks as multi-line-comment start blocks to exit multi-line-comments.

A note: making it parse as one bit comment token now, may be the easy solution, but may result in problems, should parsing of comments in a doc-block fashion be needed later. Just food for thought.

**Comment by TheDcoder** [comment]
> A note: making it parse as one bit comment token now, may be the easy solution, but may result in problems, should parsing of comments in a doc-block fashion be needed later. Just food for thought.

(I assume you mean "big") Can you elaborate on this? I can't think of any potential issues with this approach.

**Comment by genius257** [comment]
Yeah my bad :sweat_smile: i meant "big".

The issue as i see it is that secondary parsing of nested comment blocks would be needed, if a solution like 
phpDocumentor would be implemented for AutoIt3, using your parser. IDK, maybe I'm just rambling :smile:.
If it makes no sense to parse more than on big block for you, make it so :grin:

**Comment by TheDcoder** [comment]
Ah, I see. I am familiar with this type of "inline" documentation as I have used it in JavaScript. Thanks for bringing it up, but it won't cause us any trouble because our parser only deals with code, not the documentation or any kind of metadata embedded within directives or comments.

We can just add an extra layer of parsing inside comments along with parsing of directives (e.x. `#include`, `#pragma` etc.) if needed in the future, so we can indeed have one big block without any issues :smile:

**Comment by TheDcoder** [comment]
Fixed in eabf6f8 :)
